overlapping_panels/src/main/java/com/discord/panels/PanelsChildGestureRegionObserver.kt

```diff
@@ -42,39 +45,74 @@ class PanelsChildGestureRegionObserver : View.OnLayoutChangeListener {
     oldRight: Int,
     oldBottom: Int
   ) {
-    if (view != null) {
-      val coordinates = intArrayOf(0, 0)
-      view.getLocationOnScreen(coordinates)
-
-      val x = coordinates[0]
-      val y = coordinates[1]
-
-      val absoluteLeft = x + left
-      val absoluteTop = y + top
-      val absoluteRight = x + right
-      val absoluteBottom = y + bottom
-
-      viewIdToGestureRegionMap[view.id] = Rect(
-        absoluteLeft,
-        absoluteTop,
-        absoluteRight,
-        absoluteBottom
-      )
+    val coordinates = intArrayOf(0, 0)
+    view.getLocationInWindow(coordinates)
+
+    val x = coordinates[0]
+    val y = coordinates[1]
+
+    val absoluteRight = x + right
+    val absoluteBottom = y + bottom
 
-      publishGestureRegionsUpdate()
+    viewIdToGestureRegionMap[view.id] = Rect(
+      x,
+      y,
+      absoluteRight,
+      absoluteBottom
+    )
+
+    publishGestureRegionsUpdate()
+  }
+
+  @UiThread
+  fun register(view: View) {
+    view.addOnLayoutChangeListener(this)
+
+    val listener = ViewTreeObserver.OnScrollChangedListener {
```

Comment:
To make sure I understand this, does this mean that previously, we weren't updating the panel child gesture region on scrolls?

Reply:
yep! just on layout change.